### ISSUE TITLE
aws_kinesis_stream - cannot switch between PROVISIONED and ON_DEMAND mode

### ISSUE THREAD

**Issue by gaboraron** [hashicorp/terraform-provider-aws#22921]
### What is the problem?
Switching between PROVISIONED and ON_DEMAND capacity mode fails.

### Reproduction Steps 
Having an already deployed aws_kinesis_stream resource with 
```
shard_count = 1
stream_mode_details {
    stream_mode = "PROVISIONED"
}
```
change it to 
```
shard_count = null
stream_mode_details {
    stream_mode = "ON_DEMAND"
}
```
and try to deploy.

### What did you expect to happen?
Deployment succeed and the Kinesis stream capacity mode is updated accordingly.

### What actually happened?
The deployment failed with the following error message:

> Error: shard_count must not be set when stream_mode is ON_DEMAND

### Versions
terraform: 1.0.9
hashicorp/aws: v3.63.0

### Other information
Switching between ON_DEMAND and PROVISIONED also fails. This time claiming that no shard_count is specified even though it is.

**Comment by lvthillo** [comment]
What if you remove the line `shard_count = null` when you switch to on_demand?

**Comment by gaboraron** [comment]
It behaves the same in all the three cases:
1. shard_count = null
2. shard_count = 1
3. not specified at all

**Comment by Aanap** [comment]
I have the same issue on my environment.

**Comment by fratcliffe** [comment]
@gaboraron the `stream_mode_details` option seems not to be supported in the version of the aws provider you are using - the earliest version where it's available is [3.69.0](https://registry.terraform.io/providers/hashicorp/aws/3.69.0/docs/resources/kinesis_stream#stream_mode_details-configuration-block)

**Comment by gaboraron** [comment]
I tested recently with aws provider 4.0.0 but same behavior.
Any update on this bug?

**Comment by karthiknaren** [comment]
I tested recently with 5.9.0 AWS provider version. 
looks like the issue has not been resolved, could you please suggest 
@justinretzolk @lvthillo @fratcliffe

**Comment by ewbankkit** [comment]
@karthiknaren I am unable to reproduce this problem with recent Terraform AWS Provider versions -- are you still seeing the issue?

**Comment by BehnazDiconium** [comment]
@ewbankkit yes i still have the problem ,, terraform cannot change the mode from provisioned to on-demand. I had to manually do it from console (using version = ">= 5.2.0, < 5.94.2" for aws provider)

**Comment by pjgjordaan** [comment]
Still having this issue with provider  6.14.1
